src/complex.rb

```diff
@@ -32,7 +32,7 @@ def to_s
     if real.respond_to?(:nan?) && real.nan?
       s << "NaN"
     else
-      s << real.inspect
+      s << real.to_s
```

Comment:
This looks like the reversal of #1223, where I fixed `Complex#inspect`, so I guess this change is going to work for `Complex#to_s` but break `Complex#inspect`.

I think `real.send(__method__)` will work (and the same change on line 55)

Reply:
Kind of my bad as well, when I fixed `Complex#inspect` and forgot to check `Complex#to_s`.

Reply:
That's probably the best solution. You could work around with Fiber storage (similar to `Set#inspect`), but I'm not sure if that's going to be much better.

Reply:
Done, not ideal but it works

Reply:
Another option that looks to be working:
```ruby
to_s_wrapper = lambda do |obj, method_name|
  real = obj.real
  imaginary = obj.imaginary
  # From here it's pretty much just the old to_s implementation
end

# Need to use define_method to create a closure to access to_s_wrapper
define_method(:to_s) do 
  to_s_wrapper.call(self, :to_s)
end
```
This makes the code a bit more convoluted, but at least it doesn't show up in the private methods of the object